src/services/dogs.js

```diff
@@ -0,0 +1,11 @@
+import { checkError, client } from './client';
+
+export async function fetchDogs() {
+  const resp = await client.from('dogs').select().order('name');
+  return checkError(resp);
+}
+
+export async function fetchDogById(id) {
+  const resp = await client.from('dogs').select('*').match({ id });
```

Comment:
if you .single() here, supabase will remove the array that .match() returns so you don't have to [0] it later.